aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,732 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can render templated HTML with looping control statements. For example, to render a list of people:
+
+.. code-block:: none
+
+  @{
+      var people = new Person[]
+      {
+            new Person("John", 33),
+            new Person("Doe", 41),
+      };
+  }
+
+``@for``
+^^^^^^^^^
+
+.. code-block:: none
+
+  @for (var i = 0; i < people.Length; i++)
+  {
+      var person = people[i];
+      <p>Name: @person.Name</p>
+      <p>Age: @person.Age</p>
+  }
+
+``@foreach``
+^^^^^^^^^^^^
+
+.. code-block:: none
+
+  @foreach (var person in people)
+  {
+      <p>Name: @person.Name</p>
+      <p>Age: @person.Age</p>
+  }
+
+``@while``
+^^^^^^^^^^^^
+
+.. code-block:: none
+
+  @{ var i = 0; }
+  @while (i < people.Length)
+  {
+      var person = people[i];
+      <p>Name: @person.Name</p>
+      <p>Age: @person.Age</p>
+
+      i++;
+  }
+
+``@do while``
+^^^^^^^^^^^^^^^^
+
+.. code-block:: none
+
+  @{ var i = 0; }
+  @do
+  {
+      var person = people[i];
+      <p>Name: @person.Name</p>
+      <p>Age: @person.Age</p>
+
+      i++;
+  } while (i < people.Length);
+
+Compound ``@using``
+^^^^^^^^^^^^^^^^^^^^
+
+Compound using statements can be used to represent scoping. For instance, we can utilize :doc:`/mvc/views/html-helpers` to render a form tag with the ``@using`` statement:
+
+.. code-block:: none
+
+  @using (Html.BeginForm())
+  {
+      // Form content.
+  }
+
+You can also perform scope level actions like the above with :doc:`/mvc/views/tag-helpers/index`.
+
+``@try``, ``catch``, ``finally``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Exception handling is similar to  C#:
+
+.. literalinclude:: razor/sample/Views/Home/Contact7.cshtml
+  :language: html
+
+``@lock``
+^^^^^^^^^
+
+Razor has the capability to protect critical sections with lock statements:
+
+.. code-block:: none
+
+  @lock (SomeLock)
+  {
+      // Do critical section work
+  }
+
+Comments
+^^^^^^^^^^
+
+Razor supports C# and HTML comments. The following markup:
+
+.. code-block:: none
+
+  @{
+      /* C# comment. */
+      // Another C# comment.
+  }
+  <!-- HTML comment -->
+
+Is rendered by the server as:
+
+.. code-block:: none
+
+  <!-- HTML comment -->
+
+Razor comments are removed by the server before the page is rendered. Razor uses ``@*  *@`` to delimit comments. The following code is commented out, so the server will not render any markup:
+
+.. code-block:: none
+
+    @*
+    @{
+        /* C# comment. */
+        // Another C# comment.
+    }
+    <!-- HTML comment -->
+   *@
+
+.. _razor-directives-label:
+
+Directives
+-----------
+Razor directives are represented by implicit expressions with reserved keywords following the ``@`` symbol. A directive will typically change the way a page is parsed or enable different functionality within your Razor page. A Razor page is just a generated C# file. A simple example of what a Razor page generates:
```

Comment:
> A Razor page is just a generated C# file

Start a new paragraph by moving up some of the text that you have later, like this:

Understanding how Razor generates code for a view will make it easier to understand how directives work.  A Razor page is used to generate a C# file. For example, this Razor page:

```
@{
    var output = "Hello World";
}

<div>Output: @output</div>
```

Generates this C# file:

```c#
public class MyRazorPage : RazorPage<dynamic>
{
    public override Task ExecuteAsync()
    {
        var output = "Hello World";

        WriteLiteral("<div>Output: ");
        Write(output);
        WriteLiteral("</div>");

        return Task.CompletedTask;
    }
}
```